pkg/controller/controller_utils.go

```diff
@@ -973,6 +973,20 @@ func compareMaxContainerRestarts(pi *v1.Pod, pj *v1.Pod) *bool {
 	return nil
 }
 
+func FilterClaimedPods(controller metav1.Object, selector labels.Selector, pods []*v1.Pod) []*v1.Pod {
```

Comment:
document that this limits to pods that match the selector AND are controlled by the controller